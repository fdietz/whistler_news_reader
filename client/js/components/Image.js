import React, { PropTypes, Component } from 'react';
import classNames from 'classnames';

class Image extends Component {

  static propTypes = {
    imageUrl: PropTypes.string.isRequired,
    className: PropTypes.string,
    style: PropTypes.object,
  }

  constructor(props) {
    super(props);

    this.state = { loaded: false };
    this.onImageLoad = this.onImageLoad.bind(this);
  }

  componentDidMount() {
    const { imageUrl } = this.props;
    const img = new window.Image();
    img.onload = () => this.onImageLoad();
    img.src = imageUrl;
  }

  onImageLoad() {
    const imgTag = this.imageRef;
    const { imageUrl } = this.props;
    imgTag.src = imageUrl;

    this.setState({ loaded: true });
  }

  render() {
    const { className, style, imageUrl } = this.props;
    const cls = classNames('image', className, {
      'image-loaded': this.state.loaded,
    });

    return (
      <img ref={(r) => { this.imageRef = r; }} alt={imageUrl} className={cls} style={style} />
    );
  }
}

export default Image;
