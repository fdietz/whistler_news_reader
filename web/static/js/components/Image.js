import React, { PropTypes, Component } from "react";
import ReactDOM from "react-dom";
import classNames from "classnames";

class Image extends Component {

  static propTypes = {
    src: PropTypes.string.isRequired,
    className: PropTypes.string,
    style: PropTypes.object
  }

  constructor(props) {
    super(props);

    this.state = { loaded: false };
  }

  componentDidMount() {
    const imgTag = ReactDOM.findDOMNode(this.refs.img);
    const src = imgTag.getAttribute("src");

    const img = new window.Image();
    img.onload = this.onImageLoad;
    img.src = src;
  }

  onImageLoad() {
    this.setState({ loaded: true });
  }

  render() {
    const { className, style, ...rest } = this.props;
    const cls = classNames("image", className, {
      "image-loaded": this.state.loaded
    });

    return (
      <img ref="img" className={cls} style={style} {...rest}/>
    );
  }
}

export default Image;
