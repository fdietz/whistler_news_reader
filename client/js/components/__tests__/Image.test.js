import renderer from 'react-test-renderer';
import React from 'react';
import Image from '../Image';

describe('Image', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Image imageUrl="hello" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
