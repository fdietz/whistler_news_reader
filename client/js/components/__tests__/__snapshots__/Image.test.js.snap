exports[`Image renders correctly 1`] = `
<img
  alt="hello"
  className="image"
  style={undefined} />
`;
